x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -234,12 +235,16 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
         version: this.context.env.packageInfo.version,
         ml: plugins.ml,
         lists: plugins.lists,
+        errorService: core.getStartServices().then(async ([coreStart]) => {
+          return coreStart.savedObjects.createInternalRepository();
```

Comment:
This looks to be the internal system user being used which I take it we have to use because if the API keys are destroyed for some reason we cannot report an error without using the internal Kibana user to report the errors back into the saved object?

Reply:
Yes this is correct. Utilizing the internal kibana user to ensure we can access the rules' status saved objects since we the service provided by alerting will be inaccessible.